flask_jwt_extended/tokens.py

```diff
@@ -67,18 +77,25 @@ def encode_access_token(identity, secret, algorithm, expires_delta, fresh,
         'type': 'access',
     }
 
+    # Make sure additional_token_data is in conflict with default claims
+    _check_claims(['fresh', 'type', identity_claim_key], additional_claims)
+
     # Don't add extra data to the token if user_claims is empty.
     if user_claims:
         token_data[user_claims_key] = user_claims
 
+    # Make sure additional claims is a dict before merge
+    if additional_claims and isinstance(additional_claims, dict):
```

Comment:
I think an error should be raised here instead of silently doing nothing if additional claims is not a dict.